packages/dashboard/src/components/iot-dashboard/iot-dashboard-internal.tsx

```diff
@@ -146,20 +166,86 @@ export class IotDashboardInternal {
    *
    */
 
-  onGestureStart(event: MouseEvent) {
+  onGestureStart(event: PointerEvent) {
     const { x, y } = getDashboardPosition(event);
-    const isMoveGesture = !event.shiftKey && this.isPositionOnWidget({ x, y });
-    const isSelectionGesture = event.button === MouseClick.Left;
+    const isMoveGesture =
+      !event.shiftKey && (this.isPositionOnWidget({ x, y }) || this.isPositionOnSelection({ x, y }));
+
+    const isAgainstResizeTarget =
+      event.target &&
+      (event.target as HTMLElement).matches &&
+      (event.target as HTMLElement).matches('.selection-box-corner, .selection-box-side');
+
+    if (this.activeGesture === 'resize' || isAgainstResizeTarget) {
+      // NOTE: Resize is initiated within the `<iot-selection-box />`
+      return;
+    }
 
     if (isMoveGesture) {
       this.onMoveStart({ x, y });
-    } else if (isSelectionGesture) {
-      this.onSelectionStart(event);
+    } else {
+      this.onSelectionStart(event as PointerEvent);
+    }
+  }
+
+  onPressStart(event: PointerEvent) {
+    if (event.button !== MouseClick.Left) return;
+
+    const isUnionSelection = event.shiftKey;
+
+    const { x, y } = getDashboardPosition(event);
+    const isOnSelection = this.isPositionOnSelection({ x, y });
+
+    // context menu uses a click handler, so mousedown will fire before we can cancel the event.
+    const isAgainstContextMenuTarget =
```

Comment:
is it possible for the dashboard not to know about every single possible thing that the events are trying to achieve, and instead have something else just prevent the event from propagating? I.e. the resize box could capture the event and prevent onPressStart from ever initiating

Reply:
Ya, I agree, this has taken on too much responsibility. Will refactor it